### DISCUSSION TITLE
Return the matched path/regex

### DISCUSSION THREAD

**Discussion by chris-hinds-cko** [pillarjs/path-to-regexp#296] [General]
How can I return the regex path that was matched?

I.e. if I path an array of possible paths e.g. `["/foo/:id", "/bar/:id]`

and then I call `match(["/foo/:id", "/bar/:id])`

Can I figure out which of these paths was the matched one? I thought this is what the `index` in the return was referring too but that index always seems to be `0`

**Comment by blakeembrey** [comment]
The library doesn't keep the initial input around at all, so this isn't possible today. You will need to write a small wrapper for yourself. Since it's all compiled into a regex, I don't think we even have the information in the return that'll tell you which one matched. But a simple `for` loop would work and allow you to keep the input strings around.
